src-plugins/itkDataImage/datas/itkDataImage.h

```diff
@@ -62,8 +63,18 @@ class ITKDATAIMAGEPLUGIN_EXPORT itkDataImage: public medAbstractTypedImageData<D
         return medAbstractDataFactory::instance()->registerDataType<itkDataImage<DIM,T,ID> >();
     }
 
-    // Inherited slots (through virtual member functions).
+    virtual itkDataImage* clone() {
+        auto clone = new itkDataImage(*this);
+        // create the filter that duplicates an image
+        typedef typename PrivateMember::ImageType ImageType;
+        auto duplicator = itk::ImageDuplicator<ImageType>::New();
+        duplicator->SetInputImage(d->image);
+        duplicator->Update();
+        clone->setData(duplicator->GetOutput());
+        return clone;
+    }
```

Comment:
A bit more work is needed to make the copy work right. The call to ```new itkDataImage(*this)``` will use the copy constructors, which need to be implemented for the following classes because they have pointer members that will otherwise be shallow-copied (the two separate instances will share the same ```d``` pointer for example):

```medAbstractData```:
The private member (d) needs to be deep copied, and the dataIndex should probably be reset.

```medAbstractImageData```:
The user-defined copy constructor exists but oddly does nothing, so it needs to be finished.

 ```itkDataImage```:
The private member needs to be copied. There are are several types of private classes for ```itkDataImage``` (see _itkDataPrivateTypes.h)_ and user-defined copy constructors are needed for ```itkDataImagePrivateTypeBase``` (this is where the duplicator code above should be) and ```itkDataScalarImagePrivateType```.